sechub-scan/src/main/java/com/mercedesbenz/sechub/domain/scan/product/AbstractProductExecutor.java

```diff
@@ -0,0 +1,308 @@
+import com.mercedesbenz.sechub.sharedkernel.execution.SecHubExecutionContext;
+import com.mercedesbenz.sechub.sharedkernel.execution.SecHubExecutionException;
+import com.mercedesbenz.sechub.sharedkernel.resilience.ResilientActionExecutor;
+
+/**
+ * An abstract product executor implementation
+ *
+ * @author Albert Tregnaghi
+ *
+ */
+public abstract class AbstractProductExecutor implements ProductExecutor {
+
+    private static final Logger LOG = LoggerFactory.getLogger(AbstractProductExecutor.class);
+
+    protected final ResilientActionExecutor<ProductResult> resilientActionExecutor;
+
+    @Autowired
+    protected NetworkTargetResolver targetResolver;
+
+    private ScanType scanType;
+
+    private ProductIdentifier productIdentifier;
+
+    private SecHubDataConfigurationObjectInfoFinder configObjectFinder;
+
+    private int version;
+
+    protected AbstractProductExecutor(ProductIdentifier productIdentifier, int version, ScanType scanType) {
+        if (scanType == null) {
+            throw new IllegalArgumentException("Scan type may not be null!");
+        }
+        if (productIdentifier == null) {
+            throw new IllegalArgumentException("Product identifier may not be null!");
+        }
+        this.scanType = scanType;
+        this.productIdentifier = productIdentifier;
+        this.version = version;
+
+        /*
+         * we create here our own resilient action executor instance - only for this
+         * executor instance - without spring injection
+         */
+        this.resilientActionExecutor = new ResilientActionExecutor<>();
+        this.configObjectFinder = new SecHubDataConfigurationObjectInfoFinder();
+    }
+
+    public final int getVersion() {
+        return version;
+    }
+
+    public final ProductIdentifier getIdentifier() {
+        return productIdentifier;
+    }
+
+    @Override
+    public ScanType getScanType() {
+        return scanType;
+    }
+
+    @Override
+    public final List<ProductResult> execute(SecHubExecutionContext context, ProductExecutorContext executorContext) throws SecHubExecutionException {
+        UUIDTraceLogID traceLogId = context.getTraceLogId();
+        SecHubConfiguration config = context.getConfiguration();
+
+        if (!canExecute(config)) {
+            LOG.debug("Cannot execute given configuration {}", traceLogId);
+            return Collections.emptyList();
+        }
+
+        ProductExecutorData data = createExecutorData(context, executorContext, traceLogId);
+
+        configureSourceCodeHandlingIfNecessary(data);
+        configureNetworkTargetHandlingIfNecessary(data);
+
+        return startExecution(data);
+    }
+
+    private void configureSourceCodeHandlingIfNecessary(ProductExecutorData data) {
+        if (scanType != ScanType.CODE_SCAN) {
+            return;
+        }
+        // the information about pathes is interesting for debugging but also necessary
```

Comment:
```suggestion
        // the information about paths is interesting for debugging but also necessary
```

Reply:
I will change your found parts here inside this PR. But there more than 300 of such findings and I do not want to touch too many other files.

So I decided to create https://github.com/mercedes-benz/sechub/issues/1228 for the additional ones.